### PR TITLE
Added a coordinate frame.

### PR DESCRIPTION
**Pull request by Official12345**
Added a coordinate frame for use by location detector.

### CODE REVIEW

File: code/headers/frame_types.hpp [hunk 1]
```diff
@@ -240,10 +241,28 @@ namespace r2d2 {
...
+     * 
+     * Location Detector wiki: 
+     * https://github.com/R2D2-2019/R2D2-2019/wiki/Location-Detector
+     */ 
+    struct frame_coordinate_s {
+        // This is the latitude of the coordinate. This value is always positive.
+        uint32_t latitude;
+        // This is the longitude of the coordinate. This value is always positive.
+        uint32_t longtitude;
+        // This variable describes if the latitude value is towards north or south and if the longitude value is towards east or west.
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp]: This comment is a bit on the long side for a single line

File: code/headers/frame_types.hpp [hunk 2]
```diff
@@ -240,10 +241,28 @@ namespace r2d2 {
...
+        uint8_t north_east;
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp]: So this is two directions in a single uint8_t? Maybe a name change to simething with the word direction in it is better.

**Review by itzandroidtab — CHANGES_REQUESTED** [review]
Master needs to be pulled in this repo as well.

File: code/headers/frame_types.hpp (hunk 2 above)
Comment (Official12345) [reply code/headers/frame_types.hpp]: This variable is used for 2 bools. One bool is for north or south and the other for east or west. So i can't think of a better name.

Comment (PatrickDekker98) [reply code/headers/frame_types.hpp]: Consider:
```c++
bool north_south
bool east_west
```

Comment (Official12345) [reply code/headers/frame_types.hpp]: We combined them in order to save space, but if it is fine to make it into 2 seperate bools, then i'll do that instead.

File: code/headers/frame_types.hpp [hunk 3]
```diff
@@ -240,10 +241,28 @@ namespace r2d2 {
         uint32_t voltage;
     };
 
+    /**
+     * Struct that represents a coordinate on the planet.
+     * 
+     * Location Detector wiki: 
+     * https://github.com/R2D2-2019/R2D2-2019/wiki/Location-Detector
+     */ 
+    struct frame_coordinate_s {
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp:526]: Can you change the order of the variables on size to highest to lowest.

And add a 

```cpp
#pragma pack(1)
```

before the struct

Comment (itzandroidtab) [reply code/headers/frame_types.hpp:526]: This removes the padding from the struct and saves at least 1 byte

Comment (LRstudentHU) [reply code/headers/frame_types.hpp:526]: Please add R2D2_PACK_STRUCT instead of `#pragma pack(1)`, as that is the new standard

Comment (JasperOosterbroek) [reply code/headers/frame_types.hpp:526]: Why should he change the order from highest to lowest size, does this improve performance? I think the way it is now grouped by latitude and longitude is better for readability

Comment (itzandroidtab) [reply code/headers/frame_types.hpp:526]: with pragma pack(1) you want all the big variables at the top to prevent misalignment on larger types

Comment (itzandroidtab) [reply code/headers/frame_types.hpp:526]: and this is the standard now.

File: code/headers/frame_types.hpp [hunk 4]
```diff
@@ -375,6 +376,49 @@ namespace r2d2 {
...
+     * https://github.com/R2D2-2019/location_detector
+     */
+
+    R2D2_PACK_STRUCT
+    struct frame_coordinate_s {
+        // This variable represents the height relative to the average sea level.
+        int16_t altitude;
+        
+        // This variable represents the thousandths seconds of the longitude coordinate.
+        uint16_t longitude_thousandth_second;
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp]: Maybe it is a good idea to use the short form of latitude/longitude.

https://acronyms.thefreedictionary.com/Latitude%2FLongitude

Comment (itzandroidtab) [reply code/headers/frame_types.hpp]: Same with seconds/degrees/minutes

Comment (Gian1080) [reply code/headers/frame_types.hpp]: This makes it more clear for the end user, a long could also be a datatype which could cause confusion. Min is a protected keyword.

File: code/headers/frame_types.hpp [hunk 5]
```diff
@@ -375,6 +376,49 @@ namespace r2d2 {
...
+        uint8_t longitude_degrees;
+        
+        // This variable represents the minutes of the longitude coordinate.
+        uint8_t longitude_minutes;
+        
+        // This variable represents the seconds of the longitude coordinate.
+        uint8_t longitude_seconds;
+        
+        // This variable represents the nothern or southern hemisphere the coordinate is located on.
+        char north_south_hemisphere;
```
Comment (LRstudentHU) [inline code/headers/frame_types.hpp]: Why use a char instead of a bool?

**Review by LRstudentHU — CHANGES_REQUESTED** [review]

File: code/headers/frame_types.hpp (hunk 5 above)
Comment (Official12345) [reply code/headers/frame_types.hpp]: Quite a lot of people told me it was more readable

Comment (itzandroidtab) [reply code/headers/frame_types.hpp]: It should be a bool if you are using it as a bool. A char is for characters.

Comment (Official12345) [reply code/headers/frame_types.hpp]: I've been told that it is way more clear if you use 's' for south for example

**Review by LRstudentHU — APPROVED** [review]

**Review by itzandroidtab — APPROVED** [review]

**Pull request merged**